source/edge-server/permissions.txt

```diff
@@ -0,0 +1,266 @@
+       "write": {
+         "facility_id": "%%user.id"
+       },
+       "read": {
+         "facility_id": "%%user.id"
+       },
+     },
+     "read": true,
+     "write": true,
+     "insert": true,
+     "delete": true,
+     "search": true
+   }
+
+In this example, every document has a ``facility_id`` property whose value is
+the Edge Server's ``user_id``. This role means that the Edge Server can only
+read and write documents with a ``facility_id`` matching its own ``user_id``.
+
+Read All, Write Own
++++++++++++++++++++
+
+You can configure an Edge Server instance to be able to read all data, but
+write only its own data. For example, in a retail setting, an Edge Server 
+instance might represent a store inventory system. You might want the store to
+have the ability to search across all inventory for an item, but only sell 
+items in its own inventory.
+
+.. code-block:: json
+
+   {
+     "name": "readAllWriteOnlyStoreItems",
+     "apply_when": {},
+     "document_filters": {
+       "write": {
+         "store_id": "%%user.id"
+       },
+       "read": true
+     },
+     "read": true,
+     "write": true,
+     "insert": true,
+     "delete": true,
+     "search": true
+   }
+
+Client Permissions
+~~~~~~~~~~~~~~~~~~
+
+Clients that connect to an Edge Server instance have their own permissions. 
+You can think of an Edge Server instance's permissions as a filter between the
+data in Atlas and the data that a client can read and write.
+
+In a healthcare setting similar to the Edge server instance example above, the
+Edge Server instance may only read and write data that is relevant to its own 
+facility. A client may then further restrict the permissions. For example, a 
+patient in the facility may only view their own data.
+
+You represent these as independent roles within the ``roles`` array.
+For example:
+
+.. code-block:: json
+
+   {
+     "collection": "Visits",
+     "database": "PatientRecords",
+     "roles": [
+       {
+         "name": "facilityItemsOnly",
+         "apply_when": { 
+           "%%user.type": {
+             "$eq": "edge"
+           }
+         },
+         "document_filters": {
+           "write": {
+             "facility_id": "%%user.id"
+           },
+           "read": {
+             "facility_id": "%%user.id"
+           },
+         },
+         "insert": true,
+         "delete": true,
+         "search": true,
+         "read": true,
+         "write": true
+       },
+       {
+         "name": "patientOwnRecordsOnly",
+         "apply_when": {},
+         "document_filters": {
+           "write": {
+             "patient_id": "%%user.id"
+           },
+           "read": {
+             "patient_id": "%%user.id"
+           }
+         },
+         "read": true,
+         "write": true,
+         "insert": true,
+         "delete": true,
+         "search": true
+       }
+     ]
+   }
+
+In this example, a ``PatientRecords`` database has a ``Visits`` collection.
```

Comment:
same as earlier - I'd bump this above the code. might just be me, but I find it helpful to have the context for the code example first. otherwise I have to re-read the code after reading the explanation